nncf/torch/knowledge_distillation/knowledge_distillation_loss.py

```diff
@@ -46,6 +46,11 @@ def kd_loss_fn(ref_output: torch.Tensor, compressed_model_output: torch.Tensor):
         else:
             def kd_loss_fn(ref_output: torch.Tensor, compressed_model_output: torch.Tensor):
                 mse = torch.nn.MSELoss()
+                if len(ref_output.shape) < 2:
+                    nncf_logger.debug("Incompatible shape (compressed - {}, ref - {}) of the model output tensor "
```

Comment:
It would be more clear, if error message said about number of dimensions. Also it might be not clear for the user what is `ref`. Probably there should be `student` and `teacher`

Reply:
Making this message a warning instead of debug info will make it appear every train step and console logging will not be readable. So I suggest to let it stay as a debug info.

WARNING:nncf:Incompatible number of dimensions of the model output tensor for softmax KD (student - torch.Size([]), teacher - torch.Size([]), number of dims 0 should be == 2) - ignoring!
WARNING:nncf:Incompatible number of dimensions of the model output tensor for softmax KD (student - torch.Size([]), teacher - torch.Size([]), number of dims 0 should be == 2) - ignoring!
  9%|██████████████▍                                                                                                                                            | 37/396 [00:51<08:11,  1.37s/it, loss=0.221]
WARNING:nncf:Incompatible number of dimensions of the model output tensor for softmax KD (student - torch.Size([]), teacher - torch.Size([]), number of dims 0 should be == 2) - ignoring!
WARNING:nncf:Incompatible number of dimensions of the model output tensor for softmax KD (student - torch.Size([]), teacher - torch.Size([]), number of dims 0 should be == 2) - ignoring!